Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/IDrugService.cs

```diff
@@ -14,9 +14,20 @@ public interface IDrugService
         Task UpdateDrugReportAsync(DrugReportClass drugReport, CancellationToken cancellationToken);
         Task DeleteDrugReportAsync(string reportSafetyId, CancellationToken cancellationToken);
         Task<IEnumerable<DrugReportClass>> GetAllDrugReportsAsync(CancellationToken cancellationToken);
-
         Task LogTransactionAsync(string message, CancellationToken cancellationToken);
-
         Task<IEnumerable<DrugReportClass>> SearchAndSaveDrugReportsAsync(string searchTerm, CancellationToken cancellationToken);
+
+        public delegate TResult DrugReportOperation<TResult>(DrugReportClass report1, DrugReportClass report2);
```

Comment:
The task was to use Func/Action delegate from the System namespace, but you implemented your own generic delegate instead. I accept this try because the implementation of a generic delegate is the work above the requirements.
At the same time, I just pointed out the actual requirement just to take into account